### DISCUSSION TITLE
[Bug] class[class org.apache.catalina.core.StandardHostValve] after method[invoke] intercept failure

### DISCUSSION THREAD

**Discussion by loveyang2012** [apache/skywalking#12850] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

when I used sw java agent(9.3) to trace springboot(3.2) app,I get an error like this:

ERROR 2024-12-11 18:13:46.794 http-nio-27070-exec-70 InstMethodsInter : class[cl
ass org.apache.catalina.core.StandardHostValve] after method[invoke] intercept f
ailure                                                                          
java.lang.ClassCastException: class org.apache.catalina.connector.Response canno
t be cast to class javax.servlet.http.HttpServletResponse (org.apache.catalina.c
onnector.Response and javax.servlet.http.HttpServletResponse are in unnamed modu
le of loader 'app')                                                             
        at org.apache.skywalking.apm.plugin.tomcat78x.TomcatInvokeInterceptor.af
terMethod(TomcatInvokeInterceptor.java:93)                                      
        at org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstM
ethodsInter.intercept(InstMethodsInter.java:97)                                 
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.j
ava)                                                                            
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.j
ava:93)                                                                         
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineVal
ve.java:74)                                                                     
        at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:73
1)                                                                              
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.jav
a:344)                  
at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java
:389)                                                                           
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLig
ht.java:63)                                                                     
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(Abstract
Protocol.java:896)                                                              
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpo
int.java:1741)                                                                  
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBas
e.java:52)                                                                      
        at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoo
lExecutor.java:1190)                                                            
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPo
olExecutor.java:659)                                                            
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskTh
read.java:63)                                                                   
        at java.base/java.lang.Thread.run(Thread.java:831)

### What you expected to happen

maybe agent is not work with springboot 3.2

### How to reproduce

-javaagent:/opt/skywalking-agent/skywalking-agent.jar -Dskywalking.agent.service_name=push-center-openapi -Dskywalking.collector.backend_service=xxxx

### Anything else

no

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
I don't see Spirng boot 3.2 verfied. So, it may not be supported.

**Reply by loveyang2012** [reply]
In this case, how can I configure it to avoid the above errors? I just need to collect information about the database, middleware, etc., and I already have metrics for the database.
does this ok ？
plugin.tomcat.enabled=false

**Reply by wu-sheng** [reply]
If that plugin doesn't work, no metric and tracing is not working either.

You could removd that plugin, or disable it based on the plugin name.

Your config seems not correct. I don't know where you learn from.

**Reply by wu-sheng** [reply]
Generally, we need a new plugin implementation.

**Reply by bzxiao** [reply]
The error you're encountering is probably due to mixing `javax` and `Jakarta` dependencies, which prevents the plugin from functioning properly. I seem to have encountered this situation before.
